morepath/directive.py

```diff
@@ -273,19 +275,19 @@ def predicate_key(self, reg_registry, predicate_registry):
         # dispatch_external_predicates functions that have been used,
         # or we should only allow their registration through a special
         # Morepath directive so that we can.
-        if self.func.external_predicates:
+        actual = reg_registry[self.func]
```

Comment:
I was surprised to see ``reg_registry`` here, as I thought it would be gone on your branch as it is on mine. I see you import it from a local ``dispatch`` module, which I still need to read.